car.py

```diff
@@ -0,0 +1,74 @@
+"""
+Improve previous task to make it possible to run code in this way:
+
+car = Car('BMW', 2018, 300)
+petrol_car = car.set_car_type('petrol')  # It returns `PetrolCar` instance.
+petrol_car.get_model()  # Returns `BMW`
+petrol_car.get_year()  # Returns 2018
+petrol_car.get_max_speed()  # Returns 300
+
+gas_car = car.set_car_type('gas')  # It returns `GasCar` instance.
+
+"""
+
+
+class Car:
+    def __init__(self, model, year, max_speed):
+        self.model = model
+        self.year = year
+        self.max_speed = max_speed
+
+    def set_car_type(self, new_type):
+        mapped_classes = {
```

Comment:
It's not the best idea to have mapping inside of your class.

Reply:
done

Reply:
corrected for the best option

Reply:
It's still the part of the class, but it shouldn't.

Reply:
done